packages/swapper/src/swappers/JupiterSwapper/swapperApi/getTradeQuote.ts

```diff
@@ -0,0 +1,231 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import {
+  ASSET_NAMESPACE,
+  CHAIN_NAMESPACE,
+  CHAIN_REFERENCE,
+  fromAssetId,
+  solAssetId,
+  toAssetId,
+  wrappedSolAssetId,
+} from '@shapeshiftoss/caip'
+import type { GetFeeDataInput } from '@shapeshiftoss/chain-adapters'
+import type { KnownChainIds } from '@shapeshiftoss/types'
+import { bnOrZero, convertDecimalPercentageToBasisPoints } from '@shapeshiftoss/utils'
+import type { Result } from '@sniptt/monads'
+import { Err, Ok } from '@sniptt/monads'
+import { v4 as uuid } from 'uuid'
+
+import { getDefaultSlippageDecimalPercentageForSwapper } from '../../../constants'
+import type {
+  CommonTradeQuoteInput,
+  GetTradeRateInput,
+  ProtocolFee,
+  SwapErrorRight,
+  SwapperDeps,
+  TradeQuote,
+  TradeRate,
+} from '../../../types'
+import { SwapperName, TradeQuoteError } from '../../../types'
+import { getRate, makeSwapErrorRight } from '../../../utils'
+import { getJupiterQuote, isSupportedChainId } from '../utils/helpers'
+
+const _getTradeQuote = async (
+  input: CommonTradeQuoteInput,
+  deps: SwapperDeps,
+): Promise<Result<TradeQuote[], SwapErrorRight>> => {
+  const {
+    sellAsset,
+    buyAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmount,
+    affiliateBps,
+    receiveAddress,
+    accountNumber,
+    slippageTolerancePercentageDecimal,
+    isManualReceiveAddress,
+  } = input
+
+  const { assetsById } = deps
+
+  const jupiterUrl = deps.config.REACT_APP_JUPITER_API_URL
+
+  if (isManualReceiveAddress && buyAsset.assetId === solAssetId) {
```

Comment:
it's already blocked as rate time as getTradeQuote === getTradeRate for Jupiter

I would not add any specific cases in getEnabledSwappers and isCrossAccountTradeSupported as it should belong to the swapper itself, those checks are pretty dumb, for the sake of simplicity and plumbing, I would keep specific cases into each swapper handlers instead of getting them outside of their scopes

Reply:
>  for the sake of simplicity and plumbing

Main concern comes re: plumbing this all the way through, vs. `getEnabledSwappers` taking an additional param in which would be a one-liner in two places, not confident with the amount of boilerplate this brings and possible risks vs. passing an additional prop in one function call only

Reply:
Also re: `getTradeQuote === getTradeRate`, the functions may or may not be the same, but the quotes are two different ones (rate quote at input time vs. quote quote at confirm time) - FYI once you do connect a wallet, cross-account checks will kick in and a new rate will be gotten 🙏🏽 meaning you can get into a state where you get all swappers for a given pair, but once you connect and select some cross-accout pair, the previous selection should be void.

This is part of the swapper flow as it is, and I believe we should keep it as simple as-is, the only variable here as opposed to good ol' lookup in constants for "does this swapper support cross-account trade" is the addition of the notion of `isSolanaBuyAssetId`

Reply:
Leverage `getEnabledSwappers` in [0001e53](https://github.com/shapeshift/web/pull/8120/commits/0001e5316ccd4395724ab75e36a101390df9b1d8)